### PR TITLE
Add openSCAP info, oval and xccdf scanning cases, with call function

### PR DESCRIPTION
**Pull request by wnereiz**
Add "load_security_tests_openscap" call function ([poo#37006](https://progress.opensuse.org/issues/37006)). Add base module lib/openscaptest.pm
Add tests/security/openscap/oscap_info.pm ([poo#36901](https://progress.opensuse.org/issues/36901)), tests/security/openscap/oscap_oval_scanning.pm ([poo#36904](https://progress.opensuse.org/issues/36904)), tests/security/openscap/oscap_xccdf_scanning.pm ([poo#36907](https://progress.opensuse.org/issues/36907))

- Related ticket:
  - https://progress.opensuse.org/issues/36901
  - https://progress.opensuse.org/issues/36904
  - https://progress.opensuse.org/issues/36907
  - https://progress.opensuse.org/issues/37006
- Verification run:
  - SLE: http://10.67.17.9/tests/917
  - Tumbleweed: http://10.67.17.9/tests/918

### CODE REVIEW

File: lib/openscaptest.pm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+sub pre_run_hook {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    zypper_call('in openscap-utils');
+
+    $self->oscap_get_test_file("oval.xml");
+    $self->oscap_get_test_file("xccdf.xml");
```
Comment (mitiao) [inline lib/openscaptest.pm]: I guess you are missing to submit `openscap/oval.xml` and `openscap/xccdf.xml` in this PR?
Also I think this two lines would be called in every test openscap module, that seems duplicate action. Better to add conditional statements to donwload them once, or you can clean up downloaded xml in every openscap module.

File: tests/security/openscap/oscap_oval_scanning.pm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        result="false".*
+        definition_id="oval:no_direct_root_logins:def:1".*
+        result="not\ evaluated"/sxx';
+
+    validate_script_output "oscap oval eval --results $oval_result oval.xml", sub { $scanning_match };
+    $self->validate_result($oval_result, $result_match);
+
+    validate_script_output "oscap oval eval --id oval:rule_misc_sysrq:def:1 --results $oval_result_single oval.xml", sub { $scanning_match_single };
+    $self->validate_result($oval_result_single, $result_match_single);
+
```
Comment (mitiao) [inline tests/security/openscap/oscap_oval_scanning.pm]: Please remove this empty line

File: lib/openscaptest.pm (hunk 1 above)
Comment (wnereiz) [reply lib/openscaptest.pm]: Thank you for pointing that, I just added ```data/openscap/oval.xml``` and ```data/openscap/xccdf.xml```
For __$self->oscap_get_test_file__, it has already involved a if condition with __get_test_data__. When it find these file existed, it will not re-download them.

File: tests/security/openscap/oscap_oval_scanning.pm (hunk 2 above)
Comment (wnereiz) [reply tests/security/openscap/oscap_oval_scanning.pm]: Done!

File: lib/openscaptest.pm (hunk 1 above)
Comment (mitiao) [reply lib/openscaptest.pm]: `get_test_data` check data file and returns content from `/data` directory and doesn't check if it exists in SUT. From your prove runs download executed in every openscap module.

Comment (wnereiz) [reply lib/openscaptest.pm]: Okay, I have changed the code of oscap_get_test_file to test if file existed before downloading, and updated verification test runs.
